pkg/minikube/registry/drvs/kvm2/kvm2.go

```diff
@@ -146,10 +142,40 @@ func status() registry.State {
 		}
 	}
 
+	// Allow no more than 10 seconds for connection to the hypervisor
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
 	// On Ubuntu 19.10 (libvirt 5.4), this fails if LIBVIRT_DEFAULT_URI is unset
-	cmd := exec.CommandContext(ctx, path, "domcapabilities", "--virttype", "kvm")
+	cmd := exec.CommandContext(ctx, path, "connect")
```

Comment:
@macaptain thanks for this PR, and improving kvm

I am worried this would cause a long wait for Auto-Selecting the Docker Driver

Is there a way to Check if virsh is the State of "Starting..." so only in that cause we wait for it more ?

Reply:
Hey @medyagh, thanks for the review!

We're still running the `virsh domcapabilities` line. See line 176 below. https://github.com/kubernetes/minikube/blob/29857a0026fb246d209cab1e973911d01675350c/pkg/minikube/registry/drvs/kvm2/kvm2.go#L176

The idea to run `virsh connect` was from the suggestion in the issue I raised: https://github.com/kubernetes/minikube/issues/11113#issuecomment-821263413. I don't know of a way to test whether connections to the socket are blocking or not, other than trying a connection to the socket. `libvirtd` is in active (running) state while the socket is blocking, and `virsh` blocks for around 3 seconds, but we hit a timeout in minikube after 2 seconds.

If you have concerns (and I am not sure how much value the `virsh connect` adds), I can:
1. Leave the virsh commands alone, don't run `virsh connect`, instead run `virsh domcapabilities` like normal.
2. Bump the command timeout to 6 seconds (this is the same timeout that is used for Docker and podman commands in driver verification)
3. Test for `ctx.Err() == context.DeadlineExceeded` and provide a better error message in the `registry.State` if we do hit a timeout.

This will fix my issue and will be a smaller change. Does that plan sound good? (cc @afbjorklund)

Reply:
thats fair ! what would like to know are some data we collected that 10 second is enough or thats the lowest could go ?
how did we choose 10 seconds ?

Reply:
Hey @medyagh. I've reworked the change with the plan described above.

I've chosen 6 seconds to be the timeout value in my new change because it matches the same timeout used in Docker and podman driver validation. There was no science behind 10 seconds, @afbjorklund noticed that the timeout value will be different on different systems so I picked a number that seemed high enough. For my use case, I need at least 4 seconds (I have observed the socket blocking for more than 3 seconds, but never more than 4 seconds). Let me know if you want the timeout value lowering from 6 seconds.

Reply:
sounds good